addons/account/models/account_payment.py

```diff
@@ -199,19 +199,22 @@ def _onchange_journal(self):
 
     @api.onchange('partner_id')
     def _onchange_partner_id(self):
-        if self.invoice_ids and self.invoice_ids[0].invoice_partner_bank_id:
-            self.partner_bank_account_id = self.invoice_ids[0].invoice_partner_bank_id
+        invoice_ids = self.invoice_ids.filtered(lambda x: x.company_id is False or x.company_id == self.company_id)
```

Comment:
There shouldn't be invoices without any company, it would be a bug.

Reply:
Indeed. I changed it.